src/Microsoft.ML.LightGBM/LightGbmArguments.cs

```diff
@@ -0,0 +1,404 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Text;
+using System.Reflection;
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.CommandLine;
+using Microsoft.ML.Runtime.EntryPoints;
+using Microsoft.ML.Runtime.Internal.Internallearn;
+using Microsoft.ML.Runtime.LightGBM;
+using System.Collections.Generic;
+
+[assembly: LoadableClass(typeof(LightGbmArguments.TreeBooster), typeof(LightGbmArguments.TreeBooster.Arguments),
+    typeof(SignatureLightGBMBooster), "Tree Booster", "gbdt")]
+[assembly: LoadableClass(typeof(LightGbmArguments.DartBooster), typeof(LightGbmArguments.DartBooster.Arguments),
+    typeof(SignatureLightGBMBooster), "Dropout Tree Booster", "dart")]
+[assembly: LoadableClass(typeof(LightGbmArguments.GossBooster), typeof(LightGbmArguments.GossBooster.Arguments),
+    typeof(SignatureLightGBMBooster), "Gradient-based One-Size Sampling", "goss")]
+
+[assembly: EntryPointModule(typeof(LightGbmArguments.TreeBooster.Arguments))]
+[assembly: EntryPointModule(typeof(LightGbmArguments.DartBooster.Arguments))]
+[assembly: EntryPointModule(typeof(LightGbmArguments.GossBooster.Arguments))]
+
+namespace Microsoft.ML.Runtime.LightGBM
+{
+    public delegate void SignatureLightGBMBooster();
+
+    [TlcModule.ComponentKind("BoosterParameterFunction")]
+    public interface ISupportBoosterParameterFactory : IComponentFactory<IBoosterParameter>
+    {
+    }
+    public interface IBoosterParameter
+    {
+        void UpdateParameters(Dictionary<string, string> res);
+    }
+
+    /// <summary>
+    /// Parameters names comes from LightGBM library. 
+    /// See https://github.com/Microsoft/LightGBM/blob/master/docs/Parameters.md .
```

Comment:
The actual link is `https://github.com/Microsoft/LightGBM/blob/master/docs/Parameters.rst`

Reply:
Thanks @StrikerRUS !